src/macros.jl

```diff
@@ -13,6 +13,9 @@ const CONTEXT_BINDING = Symbol("__context__")
 function generate_tag end
 function similar_context end
 
+# this @pure annotations has official vtjnash approval :p
```

Comment:
Hahaha :) we should have a linter pass for this comment

Reply:
I told @vtjnash that we should replace `@pure` with a version that expands into a call that emails him to review the code 😛